pandas/tests/test_timedelta64_overflow.py

```diff
@@ -0,0 +1,322 @@
+"""
+Check overflow behavior of operations on timedelta-valued
+DataFrames/Series/Indexes/ExtensionArrays.
+"""
+
+from itertools import (
+    chain,
+    combinations_with_replacement,
+    product,
+)
+from operator import attrgetter
+from typing import (
+    NamedTuple,
+    Type,
+    Union,
+)
+
+from hypothesis import given
+import hypothesis.strategies as st
+import pytest
+
+from pandas.errors import OutOfBoundsDatetime
+
+from pandas import (
+    DataFrame,
+    DatetimeIndex,
+    Index,
+    NaT,
+    Series,
+    Timedelta,
+    TimedeltaIndex,
+    Timestamp,
+    array,
+    to_timedelta,
+)
+import pandas._testing as tm
+from pandas.core.arrays import (
+    DatetimeArray,
+    ExtensionArray,
+    TimedeltaArray,
+)
+
+timedelta_types = (Timedelta, TimedeltaArray, TimedeltaIndex, Series, DataFrame)
```

Comment:
aside from the scalar Timedelta, this could use the box_with_array fixture in tests.arithmetic (we could plausibly move that fixture to a higher level)

Reply:
Thanks for the tip, switched to `box_with_array` in most cases. (That fixture's in the top-level `conftest`, so think we're good for now.)